vispy/visuals/filters/mesh.py

```diff
@@ -205,19 +207,17 @@ def _attach(self, visual):
     float specular_factor = 0.0;
     bool is_illuminated = diffuse_factor > 0.0;
     if (is_illuminated && shininess > 0.0) {
-        vec3 reflection = reflect(light_vec, normal);
-        specular_factor = dot(reflection, eye_vec);
-        specular_factor = max(specular_factor, 0.0);
+        // Blinn-Phong reflection
+        vec3 halfway = normalize(light_vec + eye_vec);
+        specular_factor = clamp(dot(halfway, normal), 0.0, 1.0);
```

Comment:
Changed from Phong to Blinn-Phong. If I had not done that I'd only have to change `reflect(-light_vec, normal);` (which is what you see in online code examples).

Reply:
1. As I understand it, Blinn-Phong avoids issues with Phong at certain angles, so it its is considered an improvement. I only know because I read this, so correct me if I'm wrong :)   but if this is the case it seems like a better choice then, right?
2. I see your point. I tested it by making the light_vec and eye_vec equal, but the light_vec might not mean what I thought it meant. Let's add these definitions as comments in the shader. I think I'm going to take a look at this tomorrow morning, it's too late for this now :)